dex: Add instruction for open orders initialization

armaniferrante opened the pull request:
Addresses https://github.com/project-serum/serum-dex/issues/113. Builds on https://github.com/project-serum/serum-dex/pull/112.

Additional integration tests for this exist here https://github.com/project-serum/swap/pull/1/files#diff-000b5db7bd417c0382db9381f7a340302d0c6e51564953d8111f5538eae07914R87.

sconybeare reviewed (changes requested):
Can you add the new instruction to the integration and fuzz tests? Otherwise LGTM.

armaniferrante commented:
I've rebased this and the https://github.com/project-serum/serum-dex/tree/armani/closeopen changes on master and updated the solana CLI to 1.6.9 so that the new tests would work.
